src/main/java/ch/fhnw/p2p/controller/ProjectController.java

```diff
@@ -0,0 +1,146 @@
+public class ProjectController {
+	// ------------------------
+	// PRIVATE FIELDS
+	// ------------------------
+	private Log logger = LogFactory.getLog(this.getClass());
+
+	@Autowired
+	private AccessControl accessControl;
+
+	@Autowired
+	private ProjectRepository projectRepo;
+
+	@Autowired
+	UserRepository userRepo;
+	
+	@PersistenceContext
+	EntityManager entityManager;
+
+	// ------------------------
+	// PUBLIC METHODS
+	// ------------------------
+	
+	/**
+	 * Fetch full project (for editing).
+	 */
+	@RequestMapping(value="{id}", method = RequestMethod.GET)
+	public ResponseEntity<Project> getProject(HttpServletRequest request, @PathVariable Long id) {
+		User user = accessControl.login(request, AccessControl.Allowed.COACH);
+		logger.info("GET project user=" + user.getEmail());
+		Project project = projectRepo.findById(id);
+		return new ResponseEntity<Project>(project, HttpStatus.OK);
+	}
+
+	/**
+	 * Update an existing project.
+	 */
+	@RequestMapping(value="{id}", method = RequestMethod.PUT)
+	public ResponseEntity<HttpStatus> updateProject(HttpServletRequest request, @PathVariable Long id, @Valid @RequestBody Project project, BindingResult result) {
+		User user = accessControl.login(request, AccessControl.Allowed.COACH);
+		logger.info("PUT project user=" + user.getEmail());
+		if (result.hasErrors()) {
+			return new ResponseEntity<HttpStatus>(HttpStatus.PRECONDITION_FAILED);
+		}
+		Project oldProject = projectRepo.findById(project.getId());
+		if (oldProject == null) {
+			return new ResponseEntity<HttpStatus>(HttpStatus.NOT_FOUND);
+		}
+		oldProject.setTitle(project.getTitle());
+		oldProject.setStart(project.getStart());
+		oldProject.setStop(project.getStop());
+		oldProject.setZeitmodell(project.getZeitmodell());
+		try {
+			projectRepo.save(oldProject);
+			logger.debug("Successfully updated project[" + project.getId() + "]: " + project.toString());
+			return new ResponseEntity<HttpStatus>(HttpStatus.OK);
+		} catch (Exception e) {
+			e.printStackTrace();
+			return new ResponseEntity<HttpStatus>(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * Create a new project.
+	 */
+	@RequestMapping(method = RequestMethod.POST)
+	public ResponseEntity<HttpStatus> addProject(HttpServletRequest request, @Valid @RequestBody Project project, BindingResult result) {
+		User user = accessControl.login(request, AccessControl.Allowed.COACH);
+		logger.info("POST project user=" + user.getEmail());
+		if (result.hasErrors()) {
+			return new ResponseEntity<HttpStatus>(HttpStatus.PRECONDITION_FAILED);
+		}
+		try {
+			projectRepo.save(project);
+			logger.debug("Successfully saved project[" + project.getId() + "]: " + project.toString());
+			return new ResponseEntity<HttpStatus>(HttpStatus.OK);
+		} catch (Exception e) {
+			return new ResponseEntity<HttpStatus>(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	/**
+	 * Fetch list of abbreviated projects (for overview).
+	 */
+	@RequestMapping(method = RequestMethod.GET)
```

Comment:
Aus Konsistenzgründen mit den anderen Controller wäre es schön, wenn die GET Methoden zuoberst in der Klasse stehen.